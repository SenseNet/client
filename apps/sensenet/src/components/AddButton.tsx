import { Schema } from '@sensenet/default-content-types'
import { useLogger, useRepository } from '@sensenet/hooks-react'
import {
  createStyles,
  IconButton,
  List,
  ListItem,
  ListItemIcon,
  ListItemText,
  makeStyles,
  Popover,
  Tooltip,
} from '@material-ui/core'
import { CloudUploadOutlined } from '@material-ui/icons'
import Add from '@material-ui/icons/Add'
import clsx from 'clsx'
import React, { useEffect, useState } from 'react'
import { useHistory } from 'react-router-dom'
import { globals, useGlobalStyles } from '../globalStyles'
import { useLocalization, usePersonalSettings, useQuery, useSnRoute } from '../hooks'
import { navigateToAction } from '../services'
import { useDialog } from './dialogs'
import { Icon } from './Icon'

const useStyles = makeStyles(() => {
  return createStyles({
    addButtonDisabled: {
      backgroundColor: '#CCCCCC !important',
      '& svg': {
        color: '#8C8C8C',
      },
    },
    listItem: {
      width: '100%',
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'space-evenly',
      height: globals.common.drawerItemHeight,
    },
    listDropdown: {
      padding: '10px 0 10px 10px',
      width: '245px',
      maxHeight: '548px',
    },
    listItemTextDropdown: {
      margin: 0,
      textOverflow: 'ellipsis',
      overflow: 'hidden',
      whiteSpace: 'nowrap',
      maxWidth: '139px',
    },
    disabled: {
      cursor: 'not-allowed',
    },
  })
})
export interface AddButtonProps {
  isOpened?: boolean
}

export const AddButton: React.FunctionComponent<AddButtonProps> = (props) => {
  const classes = useStyles()
  const globalClasses = useGlobalStyles()
  const repo = useRepository()
  const history = useHistory()
  const { openDialog } = useDialog()
  const [showSelectType, setShowSelectType] = useState(false)
  const [allowedChildTypes, setAllowedChildTypes] = useState<Schema[]>([])
  const localization = useLocalization().addButton
  const logger = useLogger('AddButton')
  const [isAvailable, setAvailable] = useState(false)
  const [anchorEl, setAnchorEl] = React.useState<HTMLElement | null>(null)
  const [hasUpload, setHasUpload] = useState(false)
  const personalSettings = usePersonalSettings()

  const snRoute = useSnRoute()
  const pathQuery = useQuery().get('path')
  const currentPath = `${snRoute.path ?? ''}${pathQuery ?? ''}`
  const activeAction = snRoute.match?.params.action

  useEffect(() => {
    const getActions = async () => {
      try {
        const actions = await repo.getActions({ idOrPath: currentPath })
        const isActionFound = actions.d.results.some((action) => action.Name === 'Add' || action.Name === 'Upload')
        setAvailable(isActionFound && !activeAction)
      } catch (error) {
        logger.error({
          message: localization.errorGettingActions,
          data: {
            details: { error },
          },
        })
      }
    }

    if (currentPath) {
      getActions()
    } else {
      setAvailable(false)
    }
  }, [localization.errorGettingActions, logger, repo, currentPath, activeAction])

  useEffect(() => {
    const getAllowedChildTypes = async () => {
      try {
        const allowedChildTypesFromRepo = await repo.allowedChildTypes.get({
          idOrPath: currentPath,
        })

        const filteredTypes = allowedChildTypesFromRepo.d.results
          .filter((type) => repo.schemas.getSchemaByName(type.Name).ContentTypeName === type.Name)
          .map((type) => repo.schemas.getSchemaByName(type.Name))

        const tempHasUpload = filteredTypes.some((type) => personalSettings.uploadHandlers.includes(type.HandlerName))

        setAllowedChildTypes(filteredTypes)
        setHasUpload(tempHasUpload)
      } catch (error) {
        logger.error({
          message: localization.errorGettingAllowedContentTypes,
          data: {
            details: { error },
          },
        })
      }
    }

    if (showSelectType) {
      currentPath ? getAllowedChildTypes() : setAllowedChildTypes([])
    }
  }, [
    localization.errorGettingAllowedContentTypes,
    logger,
    personalSettings.uploadHandlers,
    repo,
    showSelectType,
    currentPath,
  ])

  return (
    <div className={clsx(globalClasses.centered, globalClasses.relative)}>
      {!props.isOpened ? (
        <div className={globalClasses.drawerIconButtonWrapper}>
          {isAvailable ? (
            <Tooltip title={localization.addNew} placement="right">
              <span>
                <IconButton
                  className={globalClasses.drawerButton}
                  onClick={(event: React.MouseEvent<HTMLElement, MouseEvent>) => {
                    setAnchorEl(event.currentTarget)
                    setShowSelectType(true)
                  }}>
                  <Add className={globalClasses.drawerButtonIcon} />
                </IconButton>
              </span>
            </Tooltip>
          ) : (
            <span>
              <IconButton
                className={clsx(globalClasses.drawerButton, {
                  [classes.addButtonDisabled]: !isAvailable,
                })}
                onClick={(event: React.MouseEvent<HTMLElement, MouseEvent>) => {
                  setAnchorEl(event.currentTarget)
                  setShowSelectType(true)
                }}
                disabled={true}>
                <Add className={globalClasses.drawerButtonIcon} />
              </IconButton>
            </span>
          )}
        </div>
      ) : (
        <ListItem
          className={classes.listItem}
          button={true}
          onClick={(event: React.MouseEvent<HTMLElement, MouseEvent>) => {
            setAnchorEl(event.currentTarget)
            setShowSelectType(true)
          }}
          disabled={!isAvailable}>
          <ListItemIcon className={globalClasses.centeredHorizontal}>
            <Tooltip title={localization.addNew} placement="right">
              <span>
                <IconButton
                  className={clsx(globalClasses.drawerButtonExpanded, {
                    [classes.addButtonDisabled]: !isAvailable,
                  })}
                  disabled={!isAvailable}>
                  <Add className={globalClasses.drawerButtonIcon} />
                </IconButton>
              </span>
            </Tooltip>
          </ListItemIcon>
          <ListItemText primary={localization.addNew} />
        </ListItem>
      )}
      <Popover
        open={showSelectType}
        anchorEl={anchorEl}
        onClose={() => {
          setAnchorEl(null)
          setShowSelectType(false)
        }}
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'left',
        }}
        transformOrigin={{
          vertical: 'top',
          horizontal: 'left',
        }}>
        <List className={classes.listDropdown}>
          {hasUpload ? (
            <Tooltip title={localization.upload} placement="right">
              <ListItem
                key="Upload"
                button={true}
                style={{ padding: '10px 0 10px 10px' }}
                onClick={() => {
                  setShowSelectType(false)
                  openDialog({
                    name: 'upload',
                    props: { uploadPath: currentPath },
                    dialogProps: { open: true, fullScreen: false },
                  })
                }}>
                <ListItemIcon style={{ minWidth: '36px' }}>
                  <CloudUploadOutlined />
                </ListItemIcon>
                <ListItemText primary={localization.upload} className={classes.listItemTextDropdown} />
              </ListItem>
            </Tooltip>
          ) : null}

          {allowedChildTypes.length === 0 && !hasUpload ? (
            <div>{localization.noItems}</div>
          ) : (
            allowedChildTypes.map((childType) => (
              <Tooltip key={childType.ContentTypeName} title={childType.DisplayName} placement="right">
                <ListItem
                  button={true}
                  style={{ padding: '10px 0 10px 10px' }}
                  onClick={async () => {
                    setShowSelectType(false)
                    navigateToAction({
                      history,
                      routeMatch: snRoute.match,
                      action: 'new',
                      queryParams: { 'content-type': childType.ContentTypeName },
                    })
                  }}>
                  <ListItemIcon style={{ minWidth: '36px' }}>
                    <Icon item={childType} />
                  </ListItemIcon>
                  <ListItemText primary={childType.DisplayName} className={classes.listItemTextDropdown} />
                </ListItem>
              </Tooltip>
            ))
          )}
        </List>
      </Popover>
    </div>
  )
}
