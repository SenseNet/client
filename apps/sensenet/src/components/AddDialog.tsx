import Dialog, { DialogProps } from '@material-ui/core/Dialog/Dialog'
import DialogContent from '@material-ui/core/DialogContent'
import DialogTitle from '@material-ui/core/DialogTitle'
import { NewView } from '@sensenet/controls-react'
import { GenericContent, Schema } from '@sensenet/default-content-types'
import React, { useContext } from 'react'
import { LocalizationContext, RepositoryContext } from '../context'
import { LoggerContext } from '../context/LoggerContext'

export interface AddDialogProps {
  dialogProps: DialogProps
  schema: Schema
  parent: GenericContent
}

export const AddDialog: React.FunctionComponent<AddDialogProps> = ({ dialogProps, schema, parent }) => {
  const localization = useContext(LocalizationContext).values.addButton
  const repo = useContext(RepositoryContext)

  const handleClose = () => dialogProps.onClose && dialogProps.onClose(null as any)
  const logger = useContext(LoggerContext).withScope('AddDialog')

  return (
    <Dialog {...dialogProps}>
      <DialogTitle> {localization.dialogTitle.replace('{0}', schema.DisplayName)}</DialogTitle>
      <DialogContent>
        <NewView
          handleCancel={handleClose}
          repository={repo}
          contentTypeName={schema.ContentTypeName}
          schema={schema}
          path={parent.Path}
          title=""
          onSubmit={async (parentPath, content) => {
            try {
              const created = await repo.post({
                contentType: schema.ContentTypeName,
                parentPath,
                content,
              })
              handleClose()
              logger.information({
                message: localization.contentCreatedNotification.replace(
                  '{0}',
                  created.d.DisplayName || created.d.Name,
                ),
                data: {
                  relatedContent: created,
                  relatedRepository: repo.configuration.repositoryUrl,
                },
              })
            } catch (error) {
              logger.error({
                message: localization.errorGettingAllowedContentTypes,
                data: {
                  details: { error },
                },
              })
            }
          }}
        />
      </DialogContent>
    </Dialog>
  )
}
