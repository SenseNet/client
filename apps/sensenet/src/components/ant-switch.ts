import { Switch } from '@material-ui/core'
import { createStyles, Theme, withStyles } from '@material-ui/core/styles'

export const AntSwitch = withStyles((theme: Theme) =>
  createStyles({
    root: {
      width: 28,
      height: 16,
      padding: 0,
      display: 'flex',
    },
    switchBase: {
      padding: 2,
      color: theme.palette.common.white,
      opacity: '1',
      '&$checked': {
        transform: 'translateX(12px)',
        color: theme.palette.common.black,
        '& + $track': {
          opacity: 1,
          backgroundColor: theme.palette.common.white,
        },
      },
    },
    thumb: {
      width: 12,
      height: 12,
      boxShadow: 'none',
    },
    track: {
      opacity: 1,
    },
    checked: {},
  }),
)(Switch)
