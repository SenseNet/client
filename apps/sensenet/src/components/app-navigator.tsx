import React, { lazy, Suspense } from 'react'
import { Route, Switch } from 'react-router'
import { DesktopLayout } from './layout/DesktopLayout'
import { MainRouter } from './MainRouter'
import { FullScreenLoader } from './FullScreenLoader'

const LoginPage = lazy(() => import(/* webpackChunkName: "Login" */ './login/login-page'))

export function AppNavigator() {
  return (
    <Switch>
      <Route path="/login">
        <Suspense fallback={<FullScreenLoader />}>
          <LoginPage />
        </Suspense>
      </Route>
      <DesktopLayout>
        <MainRouter />
      </DesktopLayout>
    </Switch>
  )
}
