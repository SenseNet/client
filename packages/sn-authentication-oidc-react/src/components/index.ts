export * from './authenticating'
export * from './authentication-provider'
export * from './callback'
export * from './not-authenticated'
export * from './not-authorized'
export * from './oidc-routes'
export * from './oidc-secure'
export * from './session-lost'
export * from './silent-callback'
