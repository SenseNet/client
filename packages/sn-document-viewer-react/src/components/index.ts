export * from './comment'
export * from './document-widgets'
export * from './page-widgets'
export * from './ConfirmationDialog'
export * from './DocumentViewer'
export * from './DocumentViewerError'
export * from './DocumentViewerLayout'
export * from './DocumentViewerLoading'
export * from './DocumentViewerRegeneratePreviews'
export * from './LayoutAppBar'
export * from './Page'
export * from './PageList'
export * from './styles'
