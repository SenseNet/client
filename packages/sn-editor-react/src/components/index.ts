export * from './editor'
export * from './menu-bar'
