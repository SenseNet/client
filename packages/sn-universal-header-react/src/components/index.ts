export * from './hamburger-menu'
export * from './header'
export * from './three-dot-menu'
