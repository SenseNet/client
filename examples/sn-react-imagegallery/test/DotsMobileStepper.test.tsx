import { mount } from 'enzyme'
import React from 'react'
import { Button } from '@material-ui/core'
import { DotsMobileStepper } from '../src/components/DotsMobileStepper'

describe('handleNext', () => {
  const mock = {
    imageIndex: 1,
    steppingFunction: jest.fn(),
    imageListLenght: 1,
  }
  it('Matches snapshot', () => {
    const wrapper = mount(<DotsMobileStepper {...mock} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('Handle Next Click', () => {
    const l = mount(<DotsMobileStepper {...mock} />)
    l.find(Button)
      .last()
      .simulate('click')
    expect(mock.steppingFunction).toBeCalledWith(2, false)
  })
  it('Handle Back Click', () => {
    const l = mount(<DotsMobileStepper {...mock} />)
    l.find(Button)
      .first()
      .simulate('click')
    expect(mock.steppingFunction).toBeCalledWith(0, false)
  })
})
