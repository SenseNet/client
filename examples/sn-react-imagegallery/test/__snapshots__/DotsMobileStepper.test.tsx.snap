// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handleNext Matches snapshot 1`] = `
<Component
  imageIndex={1}
  imageListLenght={1}
  steppingFunction={[MockFunction]}
>
  <WithStyles(ForwardRef(MobileStepper))
    activeStep={1}
    backButton={
      <ForwardRef(WithStyles)
        disabled={false}
        onClick={[Function]}
        size="small"
      >
        <Memo />
        Back
      </ForwardRef(WithStyles)>
    }
    className="makeStyles-root-1"
    nextButton={
      <ForwardRef(WithStyles)
        disabled={false}
        onClick={[Function]}
        size="small"
      >
        Next
        <Memo />
      </ForwardRef(WithStyles)>
    }
    position="static"
    steps={1}
    variant="text"
  >
    <ForwardRef(MobileStepper)
      activeStep={1}
      backButton={
        <ForwardRef(WithStyles)
          disabled={false}
          onClick={[Function]}
          size="small"
        >
          <Memo />
          Back
        </ForwardRef(WithStyles)>
      }
      className="makeStyles-root-1"
      classes={
        Object {
          "dot": "MuiMobileStepper-dot",
          "dotActive": "MuiMobileStepper-dotActive",
          "dots": "MuiMobileStepper-dots",
          "positionBottom": "MuiMobileStepper-positionBottom",
          "positionStatic": "MuiMobileStepper-positionStatic",
          "positionTop": "MuiMobileStepper-positionTop",
          "progress": "MuiMobileStepper-progress",
          "root": "MuiMobileStepper-root",
        }
      }
      nextButton={
        <ForwardRef(WithStyles)
          disabled={false}
          onClick={[Function]}
          size="small"
        >
          Next
          <Memo />
        </ForwardRef(WithStyles)>
      }
      position="static"
      steps={1}
      variant="text"
    >
      <WithStyles(ForwardRef(Paper))
        className="MuiMobileStepper-root MuiMobileStepper-positionStatic makeStyles-root-1"
        elevation={0}
        square={true}
      >
        <ForwardRef(Paper)
          className="MuiMobileStepper-root MuiMobileStepper-positionStatic makeStyles-root-1"
          classes={
            Object {
              "elevation0": "MuiPaper-elevation0",
              "elevation1": "MuiPaper-elevation1",
              "elevation10": "MuiPaper-elevation10",
              "elevation11": "MuiPaper-elevation11",
              "elevation12": "MuiPaper-elevation12",
              "elevation13": "MuiPaper-elevation13",
              "elevation14": "MuiPaper-elevation14",
              "elevation15": "MuiPaper-elevation15",
              "elevation16": "MuiPaper-elevation16",
              "elevation17": "MuiPaper-elevation17",
              "elevation18": "MuiPaper-elevation18",
              "elevation19": "MuiPaper-elevation19",
              "elevation2": "MuiPaper-elevation2",
              "elevation20": "MuiPaper-elevation20",
              "elevation21": "MuiPaper-elevation21",
              "elevation22": "MuiPaper-elevation22",
              "elevation23": "MuiPaper-elevation23",
              "elevation24": "MuiPaper-elevation24",
              "elevation3": "MuiPaper-elevation3",
              "elevation4": "MuiPaper-elevation4",
              "elevation5": "MuiPaper-elevation5",
              "elevation6": "MuiPaper-elevation6",
              "elevation7": "MuiPaper-elevation7",
              "elevation8": "MuiPaper-elevation8",
              "elevation9": "MuiPaper-elevation9",
              "root": "MuiPaper-root",
              "rounded": "MuiPaper-rounded",
            }
          }
          elevation={0}
          square={true}
        >
          <div
            className="MuiPaper-root MuiPaper-elevation0 MuiMobileStepper-root MuiMobileStepper-positionStatic makeStyles-root-1"
          >
            <WithStyles(ForwardRef(Button))
              disabled={false}
              onClick={[Function]}
              size="small"
            >
              <ForwardRef(Button)
                classes={
                  Object {
                    "colorInherit": "MuiButton-colorInherit",
                    "contained": "MuiButton-contained",
                    "containedPrimary": "MuiButton-containedPrimary",
                    "containedSecondary": "MuiButton-containedSecondary",
                    "disabled": "Mui-disabled",
                    "focusVisible": "Mui-focusVisible",
                    "fullWidth": "MuiButton-fullWidth",
                    "label": "MuiButton-label",
                    "outlined": "MuiButton-outlined",
                    "outlinedPrimary": "MuiButton-outlinedPrimary",
                    "outlinedSecondary": "MuiButton-outlinedSecondary",
                    "root": "MuiButton-root",
                    "sizeLarge": "MuiButton-sizeLarge",
                    "sizeSmall": "MuiButton-sizeSmall",
                    "text": "MuiButton-text",
                    "textPrimary": "MuiButton-textPrimary",
                    "textSecondary": "MuiButton-textSecondary",
                  }
                }
                disabled={false}
                onClick={[Function]}
                size="small"
              >
                <WithStyles(ForwardRef(ButtonBase))
                  className="MuiButton-root MuiButton-text MuiButton-sizeSmall"
                  component="button"
                  disabled={false}
                  focusRipple={true}
                  focusVisibleClassName="Mui-focusVisible"
                  onClick={[Function]}
                  type="button"
                >
                  <ForwardRef(ButtonBase)
                    className="MuiButton-root MuiButton-text MuiButton-sizeSmall"
                    classes={
                      Object {
                        "disabled": "Mui-disabled",
                        "focusVisible": "Mui-focusVisible",
                        "root": "MuiButtonBase-root",
                      }
                    }
                    component="button"
                    disabled={false}
                    focusRipple={true}
                    focusVisibleClassName="Mui-focusVisible"
                    onClick={[Function]}
                    type="button"
                  >
                    <button
                      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-sizeSmall"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiButton-label"
                      >
                        <ForwardRef>
                          <WithStyles(ForwardRef(SvgIcon))>
                            <ForwardRef(SvgIcon)
                              classes={
                                Object {
                                  "colorAction": "MuiSvgIcon-colorAction",
                                  "colorDisabled": "MuiSvgIcon-colorDisabled",
                                  "colorError": "MuiSvgIcon-colorError",
                                  "colorPrimary": "MuiSvgIcon-colorPrimary",
                                  "colorSecondary": "MuiSvgIcon-colorSecondary",
                                  "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                  "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                  "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                  "root": "MuiSvgIcon-root",
                                }
                              }
                            >
                              <svg
                                aria-hidden="true"
                                className="MuiSvgIcon-root"
                                focusable="false"
                                role="presentation"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                                />
                                <path
                                  d="M0 0h24v24H0V0z"
                                  fill="none"
                                />
                              </svg>
                            </ForwardRef(SvgIcon)>
                          </WithStyles(ForwardRef(SvgIcon))>
                        </ForwardRef>
                        Back
                      </span>
                      <NoSsr>
                        <WithStyles(TouchRipple)
                          center={false}
                        >
                          <TouchRipple
                            center={false}
                            classes={
                              Object {
                                "child": "MuiTouchRipple-child",
                                "childLeaving": "MuiTouchRipple-childLeaving",
                                "childPulsate": "MuiTouchRipple-childPulsate",
                                "ripple": "MuiTouchRipple-ripple",
                                "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                "rippleVisible": "MuiTouchRipple-rippleVisible",
                                "root": "MuiTouchRipple-root",
                              }
                            }
                          >
                            <span
                              className="MuiTouchRipple-root"
                            >
                              <TransitionGroup
                                childFactory={[Function]}
                                component={null}
                                enter={true}
                                exit={true}
                              />
                            </span>
                          </TouchRipple>
                        </WithStyles(TouchRipple)>
                      </NoSsr>
                    </button>
                  </ForwardRef(ButtonBase)>
                </WithStyles(ForwardRef(ButtonBase))>
              </ForwardRef(Button)>
            </WithStyles(ForwardRef(Button))>
            2
             / 
            1
            <WithStyles(ForwardRef(Button))
              disabled={false}
              onClick={[Function]}
              size="small"
            >
              <ForwardRef(Button)
                classes={
                  Object {
                    "colorInherit": "MuiButton-colorInherit",
                    "contained": "MuiButton-contained",
                    "containedPrimary": "MuiButton-containedPrimary",
                    "containedSecondary": "MuiButton-containedSecondary",
                    "disabled": "Mui-disabled",
                    "focusVisible": "Mui-focusVisible",
                    "fullWidth": "MuiButton-fullWidth",
                    "label": "MuiButton-label",
                    "outlined": "MuiButton-outlined",
                    "outlinedPrimary": "MuiButton-outlinedPrimary",
                    "outlinedSecondary": "MuiButton-outlinedSecondary",
                    "root": "MuiButton-root",
                    "sizeLarge": "MuiButton-sizeLarge",
                    "sizeSmall": "MuiButton-sizeSmall",
                    "text": "MuiButton-text",
                    "textPrimary": "MuiButton-textPrimary",
                    "textSecondary": "MuiButton-textSecondary",
                  }
                }
                disabled={false}
                onClick={[Function]}
                size="small"
              >
                <WithStyles(ForwardRef(ButtonBase))
                  className="MuiButton-root MuiButton-text MuiButton-sizeSmall"
                  component="button"
                  disabled={false}
                  focusRipple={true}
                  focusVisibleClassName="Mui-focusVisible"
                  onClick={[Function]}
                  type="button"
                >
                  <ForwardRef(ButtonBase)
                    className="MuiButton-root MuiButton-text MuiButton-sizeSmall"
                    classes={
                      Object {
                        "disabled": "Mui-disabled",
                        "focusVisible": "Mui-focusVisible",
                        "root": "MuiButtonBase-root",
                      }
                    }
                    component="button"
                    disabled={false}
                    focusRipple={true}
                    focusVisibleClassName="Mui-focusVisible"
                    onClick={[Function]}
                    type="button"
                  >
                    <button
                      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-sizeSmall"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiButton-label"
                      >
                        Next
                        <ForwardRef>
                          <WithStyles(ForwardRef(SvgIcon))>
                            <ForwardRef(SvgIcon)
                              classes={
                                Object {
                                  "colorAction": "MuiSvgIcon-colorAction",
                                  "colorDisabled": "MuiSvgIcon-colorDisabled",
                                  "colorError": "MuiSvgIcon-colorError",
                                  "colorPrimary": "MuiSvgIcon-colorPrimary",
                                  "colorSecondary": "MuiSvgIcon-colorSecondary",
                                  "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                  "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                  "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                  "root": "MuiSvgIcon-root",
                                }
                              }
                            >
                              <svg
                                aria-hidden="true"
                                className="MuiSvgIcon-root"
                                focusable="false"
                                role="presentation"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                                />
                                <path
                                  d="M0 0h24v24H0V0z"
                                  fill="none"
                                />
                              </svg>
                            </ForwardRef(SvgIcon)>
                          </WithStyles(ForwardRef(SvgIcon))>
                        </ForwardRef>
                      </span>
                      <NoSsr>
                        <WithStyles(TouchRipple)
                          center={false}
                        >
                          <TouchRipple
                            center={false}
                            classes={
                              Object {
                                "child": "MuiTouchRipple-child",
                                "childLeaving": "MuiTouchRipple-childLeaving",
                                "childPulsate": "MuiTouchRipple-childPulsate",
                                "ripple": "MuiTouchRipple-ripple",
                                "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                "rippleVisible": "MuiTouchRipple-rippleVisible",
                                "root": "MuiTouchRipple-root",
                              }
                            }
                          >
                            <span
                              className="MuiTouchRipple-root"
                            >
                              <TransitionGroup
                                childFactory={[Function]}
                                component={null}
                                enter={true}
                                exit={true}
                              />
                            </span>
                          </TouchRipple>
                        </WithStyles(TouchRipple)>
                      </NoSsr>
                    </button>
                  </ForwardRef(ButtonBase)>
                </WithStyles(ForwardRef(ButtonBase))>
              </ForwardRef(Button)>
            </WithStyles(ForwardRef(Button))>
          </div>
        </ForwardRef(Paper)>
      </WithStyles(ForwardRef(Paper))>
    </ForwardRef(MobileStepper)>
  </WithStyles(ForwardRef(MobileStepper))>
</Component>
`;
